client.go

```diff
@@ -85,15 +81,32 @@ func (c *Client) IsDangerMode() bool {
 	return c != nil && c.cfg.DangerMode
 }
 
+// GetDefaultUsage returns the default usage value.
+func (c *Client) GetDefaultUsage() int64 {
+	if c.cfg.DefaultUsage != nil {
+		return *c.cfg.DefaultUsage
+	}
+	return 1
+}
+
 // FetchProjectQuota fetches the project quota from cache or from the quota server.
 func (c *Client) FetchProjectQuota(ctx context.Context, projectID uint64, now time.Time) (*proto.AccessQuota, error) {
+	logger := c.logger.With(
+		slog.String("op", "fetch_project_quota"),
+		slog.Uint64("project_id", projectID),
+	)
 	// fetch access quota
 	quota, err := c.quotaCache.GetProjectQuota(ctx, projectID)
 	if err != nil {
-		if !errors.Is(err, proto.ErrAccessKeyNotFound) {
-			return nil, err
+		if !errors.Is(err, proto.ErrAccessKeyNotFound) && !errors.Is(err, proto.ErrProjectNotFound) {
```

Comment:
Let's move the logger down in error branch, so we create new handler, only on errors.
```suggestion
		logger := c.logger.With(
				slog.String("op", "fetch_project_quota"),
				slog.Uint64("project_id", projectID),
			)
		if !errors.Is(err, proto.ErrAccessKeyNotFound) && !errors.Is(err, proto.ErrProjectNotFound) {
```